scrapy/commands/runspider.py

```diff
@@ -1,22 +1,24 @@
 import sys
-import os
+from os import PathLike
+from pathlib import Path
 from importlib import import_module
+from types import ModuleType
+from typing import Union
 
 from scrapy.utils.spider import iter_spider_classes
 from scrapy.exceptions import UsageError
 from scrapy.commands import BaseRunSpiderCommand
 
 
-def _import_file(filepath):
-    abspath = os.path.abspath(filepath)
-    dirname, file = os.path.split(abspath)
-    fname, fext = os.path.splitext(file)
-    if fext not in ('.py', '.pyw'):
+def _import_file(filepath: Union[str, PathLike]) -> ModuleType:
+    abspath = Path(filepath).resolve()
+    dirname = str(abspath.parent)
```

Comment:
very minor: probably it makes sense to move this line after the abspath.suffix check

Reply:
I moved it, but now I'm looking at `if dirname:` and thinking it can never fail. In both the old code and the new code, `abspath` is an absolute path to a file. Files must be contained within directories, so the absolute path will always have some parent path component. Even if you provide as input the absolute path to a path root, the parent of the root is the root itself.

The only way it could be false is if `abspath` were not an absolute path, in which case the old code would have returned an empty string and the new code would return `"."`.

```
>>> from pathlib import PurePosixPath, PureWindowsPath
>>> PurePosixPath('/').parent
PurePosixPath('/')
>>> PureWindowsPath('c:\\').parent
PureWindowsPath('c:/')
>>> import os
>>> os.path.split('/')
('/', '')
>>> os.path.split('c:\\') 
('c:\\', '')
>>> PureWindowsPath('a').parent            
PureWindowsPath('.')
>>> os.path.split('a')                     
('', 'a')
```